src/HealthChecks.AzureStorage/DependencyInjection/AzureStorageHealthCheckBuilderExtensions.cs

```diff
@@ -76,22 +78,94 @@ public static class AzureStorageHealthCheckBuilderExtensions
             TimeSpan? timeout = default)
         {
             return builder.Add(new HealthCheckRegistration(
-               name ?? AZURESTORAGE_NAME,
+               name ?? AZUREBLOB_NAME,
                sp => new AzureBlobStorageHealthCheck(blobServiceUri, credential, containerName, clientOptions),
                failureStatus,
                tags,
                timeout));
         }
 
+        /// <summary>
+        /// Add a health check for Azure Blob Storage.
+        /// </summary>
+        /// <remarks>
+        /// A <see cref="BlobServiceClient"/> service must be registered in the service container.
+        /// </remarks>
+        /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
+        /// <param name="configureOptions">Delegate for configuring the health check. Optional.</param>
+        /// <param name="name">The health check name. Optional. If <see langword="null"/> the type name 'azureblob' will be used for the name.</param>
+        /// <param name="failureStatus">
+        /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <see langword="null"/> then
+        /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
+        /// </param>
+        /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
+        /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
+        /// <returns>The specified <paramref name="builder"/>.</returns>
+        public static IHealthChecksBuilder AddAzureBlobStorage(
+            this IHealthChecksBuilder builder,
+            Action<AzureBlobStorageHealthCheckOptions>? configureOptions = default,
+            string? name = default,
+            HealthStatus? failureStatus = default,
+            IEnumerable<string>? tags = default,
+            TimeSpan? timeout = default)
+        {
+            var options = new AzureBlobStorageHealthCheckOptions();

```

Comment:
see #1356. In general case placing options object outside of `HealthCheckRegistration` and configuring it inside can cause memory leak. Not in this case but nonetheless...

Reply:
In future one can add some property like `List<>` and add some item inside so then BOOM!

Reply:
Good catch! I've moved it inside the lambda